src/aat/java/uk/gov/hmcts/ccd/datastore/tests/functional/elasticsearch/ElasticsearchCaseSearchSecurityTest.java

```diff
@@ -0,0 +1,177 @@
+        }
+
+    }
+
+    @Nested
+    @DisplayName("Case field security")
+    class CaseFieldSecurity {
+
+        @Test
+        @DisplayName("should return the case field where user role matches ACL and security classification")
+        void shouldReturnCaseField() {
+            ValidatableResponse response = searchCaseAndAssertCaseReference(asRestrictedCaseworker(false),
+                                                                            ES_FIELD_CASE_REFERENCE,
+                                                                            caseRefCaseFieldSecurityQuery);
+            response.body("cases[0].case_data." + ES_FIELD_EMAIL_ID, is(EMAIL_ID_VALUE));
+        }
+
+        @Test
+        @DisplayName("should NOT return the case field where user role has lower security classification than case field")
+        void shouldNotReturnCaseFieldForLowerSecurityClassification() {
+            ValidatableResponse response = searchCaseAndAssertCaseReference(asPrivateCaseworker(false),
+                                                                            ES_FIELD_CASE_REFERENCE,
+                                                                            caseRefCaseFieldSecurityQuery);
+            response.body("cases[0].case_data", not(hasKey(ES_FIELD_EMAIL_ID)));
+        }
+
+        //@Test
+        @DisplayName("should NOT return the case field where user role does not have read access on the field")
+        void shouldNotReturnCaseFieldForNoReadAccess() {
+            //TODO
+        }
+
+    }
+
+    @Nested
+    @DisplayName("User access security")
+    class UserAccessSecurity {
+
+        @Test
+        @DisplayName("should return the case for a solicitor role if granted access to the case")
+        void shouldReturnCase() {
+            //TODO
+        }
+
+        //@Test
+        @DisplayName("should NOT return the case for a solicitor role if not granted access to the case")
+        void shouldNotReturnCase() {
+            //TODO
+        }
+
+    }
+
+    private ValidatableResponse searchCaseAndAssertCaseReference(Supplier<RequestSpecification> asUser, String field, Object value) {
+        ValidatableResponse response = searchCase(asUser, field, value);
+        response.body("cases.size()", is(1));
+        response.body("cases[0]." + CASE_ID, is(value));
+
+        return response;
+    }
+
+    private void searchCaseAndAssertCaseNotReturned(Supplier<RequestSpecification> asUser, String field, Object value) {
+        ValidatableResponse response = searchCase(asUser, field, value);
+        response.body("cases.size()", is(0));
+    }
+
+    private ValidatableResponse searchCase(Supplier<RequestSpecification> asUser, String field, Object value) {
+        String jsonSearchRequest = createExactMatchSearchRequest(field, value);
+        return searchCase(asUser, jsonSearchRequest);
+    }
+
+    private String createExactMatchSearchRequest(String field, Object value) {
```

Comment:
we should avoid having query creation inside a test. Can we encapsulate this in a class ElasticSearchQueries?

Reply:
also, any change we can avoid working directly with Json? For example is it possible to use the Jest or ES client query builders?

Reply:
I have moved the queries to a separate class. 

Using Jest or ES client within the functional test may be a little complex from QA perspective. If we keep it as JSON queries, there will be less dependency on the developers as QAs can easily maintain JSON rather than learning and implementing ES Querybuilder APIs.